shell/src/main/java/alluxio/master/MasterHealthCheckClient.java

```diff
@@ -78,38 +115,64 @@ public Builder withConfiguration(AlluxioConfiguration alluxioConf) {
     }
 
     /**
-     * @param alluxioMasterName the Alluxio master process name
+     * @param masterType the Alluxio master type
      * @return an instance of the builder
      */
-    public Builder withAlluxioMasterName(String alluxioMasterName) {
-      mAlluxioMasterName = alluxioMasterName;
+    public Builder withAlluxioMasterType(MasterType masterType) {
+      mAlluxioMasterType = masterType;
+      return this;
+    }
+
+    /**
+     * @param policySupplier the retry policy supplier to use for the health check
+     * @return a builder utilizing the given retry policy supplier
+     */
+    public Builder withRetryPolicy(Supplier<RetryPolicy> policySupplier) {
+      mPolicySupplier = policySupplier;
       return this;
     }
 
     /**
      * @return a {@link MasterHealthCheckClient} for the current builder values
      */
     public HealthCheckClient build() {
-      return new MasterHealthCheckClient(mAlluxioMasterName, mProcessCheck, mConf);
+      return new MasterHealthCheckClient(mAlluxioMasterType, mProcessCheck, mPolicySupplier, mConf);
     }
   }
 
   /**
    * Runnable for checking if the AlluxioMaster is serving RPCs.
    */
-  public final class MasterServingCheckRunnable implements Runnable {
+  public final class MasterServingHealthCheck extends RpcPortHealthCheckClient implements Runnable {
```

Comment:
since this doesn't check an RPC port, it's strange for it to extend RpcPortHealthCheckClient

Reply:
It actually does check an RPC port. it uses the `super.isServing` method in the `run` function